website/docs/interactors/4-matchers-assertions.md

```diff
@@ -0,0 +1,228 @@
+
+:::note
+These different assertion methods are equivalents of Jest's `expect` and Cypress' `should`. When refactoring your test with Interactors, you would replace those constructs with the interactors' assertion methods.
+:::
+
+In the case of a form with textfields, you would assert the placeholder against a textfield like this:
+
+```js
+TextField({ id: 'username-id' }).has({ placeholder: 'USERNAME' });
+```
+
+The difference between this approach and using `exists()` is that `exists()` is less refined as it will succeed as long as there is at least one match. With the two textfields for the username and password, the assertion would fail on if we were to write a test like this:
+
+```js
+TextField().has({ placeholder: 'USERNAME' });
+```
+
+It would fail on account of the textfield that has the placeholder value `PASSWORD`. You therefore need to choose the assertion method that is most appropriate for your tests.
+
+Lastly, there is also the `is()` method, which works just as `has()` does. The only difference is in the semantics so that your tests can read better.
+
+For instance, if we wanted to test if a textfield is visible, `has()` would work perfectly fine; but writing the test using `is()` would look more natural like this:
+
+```js
+TextField({ id: 'username-id' }).is({ visible: true });
+```
+
+It makes more sense to say "text field _is_ visible" rather than "text field _has_ visible". Generally, we recommend using `is()` if your assertion is an adjective, and use `has()` if your assertion is a noun. For example, if we wanted to assert for `visibility`, then it would sound more natural to use `has()` instead:
+
+```js
+TextField({ id: 'username-id' }).has({ visibility: true });
+```
+
+## Matchers
+
+The methods in which we use locators and filters, as we have illustrated in the previous page, are very explicit. There may be times when some flexibility in your interactors is necessary in order to write better tests. And that's where matchers come in. <!-- the paragraph above needs to be fact checked -->
+
+If your tests are written against a simulated database, it might not be important what the randomized users' names are. Maybe you just want to assert for the successful login message to include the word 'welcome'. Or perhaps it does not matter what the name is but you still want to assert that a name is displayed. This is one of many situations where matchers would come in handy.
+
+We will start by going over the two most primitive matchers: `including()` and `matching()`.
+
+:::note
+Matchers are meant to be used for the _values_ of locators and filters and they cannot be substituted for the actual filters:
+```js
+Heading().has({ id: or('foo', 'bar') }); // good
+Heading().has({ or(id: 'foo', id: 'bar') }); // bad
+```
+:::
+
+### including, matching
```

Comment:
I think we'll have to make the two sections have a parallel structure. For instance, we'd have to separate the `Assertions` section in subsections by groups of functions just as we do here.

Reply:
I don't think it'd be possible to separate `Assertions` into sections because of the way it's written. And removing `Matchers` subsections doesn't seem like a good idea either.

Reply:
We have now split up the assertions section.